src/api/app/controllers/webui/package_controller.rb

```diff
@@ -705,29 +705,30 @@ def check_build_log_access
       return false
     end
 
+    @package_name = params[:package]
     begin
-      @package = Package.get_by_project_and_name(@project, params[:package], use_source: false,
-                                                                             follow_multibuild: true,
-                                                                             follow_project_links: true)
+      @package = Package.get_by_project_and_name(@project, @package_name, use_source: false,
+                                                                          follow_multibuild: true,
+                                                                          follow_project_links: true)
     rescue Package::UnknownObjectError
       redirect_to project_show_path(@project.to_param),
                   error: "Couldn't find package '#{params[:package]}' in " \
                          "project '#{@project.to_param}'. Are you sure it exists?"
       return false
     end
 
-    # package is nil for remote projects
+    # NOTE: @package is a String for multibuild packages
+    @package = Package.find_by_project_and_name(@project.name, Package.striping_multibuild_suffix(@package_name)) if @package.is_a?(String)
+
+    # NOTE: @package is nil for remote projects
     if @package && !@package.check_source_access?
-      redirect_to package_show_path(project: @project.name, package: @package.name),
+      redirect_to package_show_path(project: @project.name, package: @package_name),
                   error: 'Could not access build log'
       return false
     end
 
     @can_modify = User.possibly_nobody.can_modify?(@project) || User.possibly_nobody.can_modify?(@package)
```

Comment:
`User.possibly_nobody.can_modify?(nil)` is something different then `User.possibly_nobody.can_modify?(Package)`

did you check what `@can_modify` does?

Reply:
Ops, true. I moved down the `Package.find_by_project_and_name...` line, below the `can_modify?` check. This should fix it.